sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/SyncToken.java

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration.implementation;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * Due to the nature of some distributed systems real-time consistency between requests can't (or it's very hard) to be
+ * enforced implicitly. A solution is to allow protocol support in the form of multiple Synchronization Tokens.
+ * Synchronization tokens are optional.
+ * <p>
+ * Uses optional Sync-Token request/response headers will guarantee real-time consistency between different client
+ * instances and requests.
+ *
+ * @see <a href="https://github.com/Azure/AppConfiguration/blob/master/docs/REST/consistency.md"/>
+ */
+@Immutable
+public final class SyncToken {
+    private final ClientLogger logger = new ClientLogger(SyncToken.class);
+
+    private final String id;
+    private final String value;
+    private final Long sequenceNumber;
+
+    /**
+     * Create an instance of SyncToken class. Skip the sync token if the parsing failed.
+     *
+     * @param syncToken only one raw sync-token string from HTTP response header, ex.,
+     * <p>Sync-Token: {@code <id>=<value>;sn=<sn></p>}. But not
+     * <p>Sync-Token: {@code <id>=<value>;sn=<sn>,<id>=<value>;sn=<sn>}</p>
+     * Allows for better concurrency and client cache ability. The client may choose to use only token's last version,
+     * since token versions are inclusive. Not required for requests.
+     */
+    SyncToken(String syncToken) {
+        String localId = null;
+        String localValue = null;
+        Long localSequenceNumber = null;
+        if (!CoreUtils.isNullOrEmpty(syncToken)) {
```

Comment:
Shouldn't we throw if it is empty rather than constructing an invalid object.

Reply:
if the `syncToken` is invalid, we don't want to throw because the user has no control of sync token do anything to correct it. The invalid 'syncToken' will be ignored. I can log the message as a warning but still, the user can't control it. What do you think? logging for every invalid case of 'syncToken'. @conniey @srnagar

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/6675/files#r373652294

Reply:
I think you're adding additional responsibilities to this class. A syncToken class only cares about itself. It doesn't know that it lives in a pipeline policy. So if it gets invalid data, it should throw.

The pipeline policy, because it understands the response comes from the service, can handle it.